instrumentor/sources_webhooks.go

```diff
@@ -260,6 +260,14 @@ func (s *SourcesValidator) validateSourceFields(ctx context.Context, source *v1a
 		))
 	}
 
+	if source.Spec.Workload.Kind == workload.WorkloadKindNamespace && source.Spec.ReportedName != "" {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("reportedName"),
+			source.Spec.ReportedName,
+			"Namespace Source must not have a reportedName, to configure a reported name for a workload edit its Source",
```

Comment:
```suggestion
			"Reported name is not valid for Namespace sources, only valid for Workload Sources",
```

just a little clearer imo